src/sql/base/browser/ui/table/formatters.ts

```diff
@@ -11,12 +11,45 @@ export interface DBCellValue {
 	isNull: boolean;
 }
 
+/**
+ * Info for executing a command. @see azdata.ExecuteCommandInfo
+ */
+export interface ExecuteCommandInfo {
+	id: string;
+	args?: string[]
+}
+
+/**
+ * The info for a DataGrid Text Cell.
+ */
+export interface TextCellValue {
+	text: string;
+	ariaLabel: string;
+}
+
+/**
+ * The info for a DataGrid Hyperlink Cell.
+ */
+export interface HyperlinkCellValue {
+	displayText: string;
+	linkOrCommand: string | ExecuteCommandInfo;
+	ariaLabel: string;
+}
+
 export namespace DBCellValue {
 	export function isDBCellValue(object: any): boolean {
 		return (object !== undefined && object.displayValue !== undefined && object.isNull !== undefined);
 	}
 }
 
+/**
+ * Checks whether the specified object is a HyperlinkCellValue object or not
+ * @param obj The object to test
+ */
+export function isHyperlinkCellValue(obj: any | undefined): obj is HyperlinkCellValue {
+	return !!(<HyperlinkCellValue>obj)?.linkOrCommand;
```

Comment:
you can do: 

export function isHyperlinkCellValue(obj: any):boolean {
	return 'linkOrCommand' in obj;
}

Reply:
I like this way since I get typing support - so if we modify the interface such that linkOrCommand is renamed then this will show a compilation error instead of just silently checking the wrong property.

Reply:
(and for this kind of thing having the return type be `obj is <type>` is a lot better too so you can get type guard support